server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -169,18 +232,7 @@ private boolean registerRule(RulesDefinition.Rule ruleDef, Map<RuleKey, RuleDefi
       newRule = false;
     }
 
-    boolean executeUpdate = false;
-    if (mergeRule(ruleDef, rule)) {
-      executeUpdate = true;
-    }
-
-    if (mergeDebtDefinitions(ruleDef, rule)) {
-      executeUpdate = true;
-    }
-
-    if (mergeTags(ruleDef, rule)) {
-      executeUpdate = true;
-    }
+    boolean executeUpdate = mergeRule(ruleDef, rule) | mergeDebtDefinitions(ruleDef, rule) | mergeTags(ruleDef, rule);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Correct this "|" to "||". ([squid:S2178](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2178))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Feh%2FSONAR-10311&issues=AWFRf5ZjsS4rYBZqFGC3&open=AWFRf5ZjsS4rYBZqFGC3)

[ISSUE_KEY]: AWFRf5ZjsS4rYBZqFGC3